integrations/gcp/gcp_core/search/iterators.py

```diff
@@ -0,0 +1,17 @@
+from typing import Any, AsyncIterable, Callable
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from port_ocean.utils.async_iterators import stream_async_iterators_tasks
+from gcp_core.search.searches import search_all_projects
+
+
+async def iterate_per_available_project(
+    project_dependent_callable: Callable[..., AsyncIterable[Any]],
+    *args: Any,
+    **kwargs: Any,
+) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    async for projects in search_all_projects():
```

Comment:
ive seen u iterating over all projects at somewhere else
do u think we should cache that?

Reply:
The output of search_all_projects is cached